Products/ZenUtils/Utils.py

```diff
@@ -2228,3 +2230,21 @@ def getQueryArgsFromRequest(request):
         elif len(parts) == 1 and parts[0]:
             query_args[parts[0]] = True
     return query_args
+
+
+def getPasswordFields(interface):
+    passwordFields = set()
+    for iface in providedBy(interface):
+        fields = getFields(iface)
+        if not fields:
+            continue
+        for key, value in fields.iteritems():
+            if isinstance(value, Password):
+                passwordFields.add(key)
+    return passwordFields
+
+
+def maskSecureProperties(data, secure_properties=[]):
+    [data.update({prop: '*' * len(data[prop])})
+        for prop in secure_properties if data.get(prop, None)]
```

Comment:
This is considered an abuse of the list comprehension syntax.  Your intent is to modify the contents of a parameter, not to create a list.  Please rewrite this as a loop that expresses your intent more accurately.  Additionally, using a list comprehension like this creates a list object that's unused.

Reply:
Thank you for your reviewing. I rewrote function by using the loop and got rid of that comprehension.